lib/jxl/modular_test.cc

```diff
@@ -322,5 +324,22 @@ TEST(ModularTest, RoundtripLosslessCustomFloat) {
                                      /*distmap=*/nullptr, pool));
 }
 
+TEST(ModularTest, ClampedGradientTest) {
+  Rng rng(0);
+  constexpr int max_bits = 29;
+  for (size_t i = 0; i < 100000; i++) {
+    int32_t a = rng.UniformI(-(1 << max_bits), 1 << max_bits);
+    int32_t b = rng.UniformI(-(1 << max_bits), 1 << max_bits);
+    int32_t c = rng.UniformI(-(1 << max_bits), 1 << max_bits);
+    int32_t cg = ClampedGradient(a, b, c);
+    int64_t g = a + b - c;
```

Comment:
This line does not do what you think it does in the presence of int32 overflow :) -- in fact, this "simple" implementation is in fact incorrect

Reply:
Sigh, it does the arithmetic in int32 with overflow and then stores it in an int64, right?
Great, I managed to make the "obviously correct" version incorrect :)

Is this an OK solution?
```
int64_t la = a, lb = b, lc = c;
int64_t g = la + lb - lc;
```

Reply:
This should do what you want it to, so long as the compiler doesn't 'optimize' it by adding b and c first.
```c
int64_t g = static_cast<int64_t>(a) + b + c;
```

Reply:
that, or just int64_t{a}+b-c should work too.

Reply:
actually looking at the underlying differenced it should be safe, and slightly faster to use a 'reinterpret_cast' as we're going from a smaller to a larger buffer, and this should just cause the compiler to push the int32_t into a int64_t compatible register.
without any checks.